### DISCUSSION TITLE
Workout details

### DISCUSSION THREAD

**Discussion by cagnulein** [sandermvanvliet/RoadCaptain#129] [General]
Hi! First of all I would like to congrats with you about your amazing project!
I have a question about the zwift protobuf, did you check if it's possible also to collect the current workout metrics from the protobuf? I mean for example target wattage and target cadence? I know that these data are available from bluetooth for example, but it will be nice  if we can extract these from the protobuf also.
Let me know!

**Comment by sandermvanvliet** [comment]
I know it’s in there but haven’t done the analysis (mostly because I don’t need it 😅)

if you want to get the data out it’s doable but a bit of work. I have a bunch of hacked together tools to pull that data apart but you’ll most definitely want to use [ProtobufDecoder](https://github.com/sandermvabvliet/ProtobufDecoder) to analyze the individual messages. 

I’m pretty limited on time and as I’m not super interested in the performance data, it’s not something I’ll be able to spend time on in the near future. 

But I’ll happily help with sharing experiences etc 👍

**Comment by cagnulein** [comment]
hi! thanks for the fast response! i guess your repository is private? i
can't open the link.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 8 gen 2024 alle 19:37 Sander van Vliet <
***@***.***> ha scritto:

**Comment by sandermvanvliet** [comment]
Probably a typo, try this https://github.com/sandermvanvliet/ProtobufDecoder

**Comment by cagnulein** [comment]
wow amazing! I will let you know what I will discover in the next weeks
hopefully!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 9 gen 2024 alle ore 13:20 Sander van Vliet <
***@***.***> ha scritto:
